mslice/util/numpy_helper.py

```diff
@@ -9,18 +10,48 @@ def apply_with_corrected_shape(method, shape_array, array, exception):
     :param array: array that has at least one axis of the same size as shape_array
     :param exception: exception to raise if array does not have at least one axis of the same size as shape_array
     """
-    if array.ndim == 1: # make 2D
-        array = array[:, np.newaxis]
-    if array.shape[1] == 1:
-        array = np.transpose(array)
-    if shape_array.shape[0] in array.shape and shape_array.shape[0] != 1:  # array matches the wrong (leftmost) axis
-        return apply_with_swapped_axes(method, shape_array, array)
-    elif shape_array.shape[1] == array.shape[1]:
-        return method(shape_array, array)
-    else:
-        raise exception
+    try:
+        return_value = method(shape_array, array)
```

Comment:
Makes sense to me to first try the method with the existing arrays, before transforming them. This does change the results of some existing tests, see `test_add_list` in `pixel_workspace_test.py` and `histogram_workspace_test.py`.